File: server/src/main/java/net/mzouabi/ng2/server/document/model/Document.java
package net.mzouabi.ng2.server.document.model;

import net.mzouabi.ng2.server.model.AbstractEntity;

import javax.persistence.Entity;

@Entity
public class Document extends AbstractEntity {

    private static final long serialVersionUID = -6321180910534044216L;

    private String entity;
    private String desciption;
    private String type;

    public String getEntity() {
        return entity;
    }

    public Document setEntity(String entity) {
        this.entity = entity;
        return this;
    }

    public String getDesciption() {
        return desciption;
    }

    public Document setDesciption(String desciption) {
        this.desciption = desciption;
        return this;
    }

    public String getType() {
        return type;
    }

    public Document setType(String type) {
        this.type = type;
        return this;
    }
}

File: front/src/app/document/model/vo/document.ts
import {EntityBase} from "@share/model/vo/entity-base";
export class Document extends EntityBase{

  private _versione: number;
  private _entity: String;
  private _desciption: String;
  private _type: String;

  get versione(): number {
    return this._versione;
  }

  set versione(value: number) {
    this._versione = value;
  }

  get entity(): String {
    return this._entity;
  }

  set entity(value: String) {
    this._entity = value;
  }

  get desciption(): String {
    return this._desciption;
  }

  set desciption(value: String) {
    this._desciption = value;
  }

  get type(): String {
    return this._type;
  }

  set type(value: String) {
    this._type = value;
  }

}
